editor/components/block-list/block.js

```diff
@@ -60,6 +61,7 @@ export class BlockListBlock extends Component {
 
 		this.setBlockListRef = this.setBlockListRef.bind( this );
 		this.bindBlockNode = this.bindBlockNode.bind( this );
+		this.updateFootnotes = this.updateFootnotes.bind( this );
```

Comment:
Also here, ideally there should be no footnote specific logic in this component.